components/local-app/pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package telemetry
+
+import (
+	"errors"
+	"log/slog"
+	"math/rand"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/gitpod-io/local-app/pkg/prettyprint"
+	segment "github.com/segmentio/analytics-go/v3"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+)
+
+// Injected at build time
+var segmentKey = "TgiJIVvFsBGwmxbnnt5NeeDaian9nr3n"
+
+var opts struct {
+	Enabled  bool
+	Identity string
+	Version  string
+
+	client segment.Client
+}
+
+// Init initialises the telemetry
+func Init(enabled bool, identity, version string) {
+	opts.Enabled = enabled
+	if !enabled {
+		return
+	}
+
+	opts.Version = version
+	opts.Identity = identity
+
+	if segmentKey != "" {
+		opts.client = segment.New(segmentKey)
+	}
+}
+
+// DoNotTrack returns true if the user opted out of telemetry
+// Implements the https://consoledonottrack.com/ proposal.
+func DoNotTrack() bool {
+	return os.Getenv("DO_NOT_TRACK") == "1"
+}
+
+// RandomIdentity generates a random identity
+func RandomIdentity() string {
+	letters := []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890")
```

Comment:
Mixpanel has payment limit per users. It would be good if we use something like machine id which is sticky.

Reply:
We could generate the machine ID by hashing the MAC adress perhaps (using [Interface.HardwareAddr](https://pkg.go.dev/net#Interface.HardwareAddr)). That should be static

Reply:
The ID is sticky because we store it in the config and only generate a fresh one if that config isn't present.

We could base that identity on the MAC.